src/runtime_src/core/pcie/common/asd_parser.cpp

```diff
@@ -0,0 +1,654 @@
+/*
+ * Copyright (C) 2023 Xilinx, Inc
+ * Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "asd_parser.h"
+#include <iostream>
+
+namespace asd_parser {
+  namespace bpt = boost::property_tree;
+
+  static std::vector<std::string> status_map;
+  static std::vector<std::string> dma_s2mm_map;
+  static std::vector<std::string> dma_mm2s_map;
+
+  static void
+  initialize_mapping()
+  {
+    // core status
+    status_map.resize(uint32_t(core_status::XAIE_CORE_STATUS_MAX_BIT));
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_ENABLE_BIT)] = "Enable";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_RESET_BIT)] = "Reset";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_MEM_STALL_S_BIT)] = "Memory_Stall_S";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_MEM_STALL_W_BIT)] = "Memory_Stall_W";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_MEM_STALL_N_BIT)] = "Memory_Stall_N";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_MEM_STALL_E_BIT)] = "Memory_Stall_E";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_LOCK_STALL_S_BIT)] = "Lock_Stall_S";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_LOCK_STALL_W_BIT)] = "Lock_Stall_W";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_LOCK_STALL_N_BIT)] = "Lock_Stall_N";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_LOCK_STALL_E_BIT)] = "Lock_Stall_E";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_STREAM_STALL_SS0_BIT)] = "Stream_Stall_SS0";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_STREAM_STALL_MS0_BIT)] = "Stream_Stall_MS0";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_CASCADE_STALL_SCD_BIT)] = "Cascade_Stall_SCD";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_CASCADE_STALL_MCD_BIT)] = "Cascade_Stall_MCD";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_DEBUG_HALT_BIT)] = "Debug_Halt";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_ECC_ERROR_STALL_BIT)] = "ECC_Error_Stall";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_ECC_SCRUBBING_STALL_BIT)] = "ECC_Scrubbing_Stall";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_ERROR_HALT_BIT)] = "Error_Halt";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_DONE_BIT)] = "Core_Done";
+    status_map[uint32_t(core_status::XAIE_CORE_STATUS_PROCESSOR_BUS_STALL_BIT)] = "Core_Proc_Bus_Stall";
```

Comment:
Hi @stsoe, these flags are coming from aie team. I have used same names so that it will be easy to maintain. But I will change these names to  follow c++ guidelines. Thanks